autogen/agentchat/conversable_agent.py

```diff
@@ -731,6 +738,148 @@ async def a_initiate_chat(
+            }
+        )
+
+        if llm_agent and llm_agent.client is not None:
+            llm_client = llm_agent.client
+        elif self.client is not None:
+            llm_client = self.client
+        else:
+            raise ValueError("No OpenAIWrapper client is found.")
+
+        response = llm_client.create(context=None, messages=_messages, cache=self.client_cache)
+        extracted_response = llm_client.extract_text_or_completion_object(response)[0]
+        if not isinstance(extracted_response, str) and hasattr(extracted_response, "model_dump"):
+            return str(extracted_response.model_dump(mode="dict"))
+        else:
+            return extracted_response
+
+    def initiate_chats(self, chats_info: List[Dict]):
+        """Initiate chats with multiple agents.
+
+        Args:
+            chats_info (List[Dict]): a list of dictionaries containing the information of the chats.
+                Each dictionary should contain the following fields:
+                - "recipient": the recipient agent.
+                - "context": any context information.
+                    "message" needs to be provided if the `generate_init_message` method is not overridden.
+                          Otherwise, input() will be called to get the initial message.
+                    "takeaway_method" can be used to specify the method to extract the takeaway from the chat.
+                        Supported methods are "last_msg" and "reflection_with_llm".
+                        when set "last_msg", it returns the last message of the dialog as the takeaway.
+                        when set "reflection_with_llm", it returns the takeaway extracted using an llm client.
+                        "reflection_with_llm" requires the llm_config to be set in either the sender or the recipient.
+                    "takeaway_prompt" can be used to specify the prompt used to extract the takeaway when takeaway_method is "reflection_with_llm".
+                        Default is None and the following default prompt will be used when "takeaway_method" is set to "reflection_with_llm":
+                        "Identify and extract the final solution to the originally asked question based on the conversation."
+                    "carryover" can be used to specify the carryover information to be passed to this chat.
+                        If provided, we will combine this carryover with the "message" content when generating the initial chat
+                        message in `generate_init_message`. Please override the `generate_init_message` if you want to customize
+                        how to use the carryover information shall be used.
+
+        """
+        receipts_set = set()
+        for chat_info in chats_info:
+            assert "recipient" in chat_info, "recipient must be provided."
+            receipts_set.add(chat_info["recipient"])
+        assert len(receipts_set) == len(chats_info), "recipients must be different."
```

Comment:
Wait wouldn't this mean I can't repeat an agent in the series? What if I want A > B > A

Reply:
Since we are doing `initiata_chat` beneath `initiata_chats`, history will be cleared for each chat session when starting a new one. In this case, if even an agent appears multiple times, the current implementation does not carry over the chat history. So it is meaningless to repeat agents, and we are explicitly enforcing users not to do so. It is indeed a limitation, though. How about we address this in a future PR?